src/Build/Instance/RunningObjectTable.cs

```diff
@@ -50,10 +58,25 @@ public object GetObject(string itemName)
             return obj;
         }
 
-        private IMoniker CreateMoniker(string itemName)
+        private Task<IMoniker> CreateMoniker(string itemName)
```

Comment:
How often will this method get called? This signature requires that you allocate a new `Task<T>` object for every invocation just to wrap the result that you mentioned you expect to normally be executed synchronously on the caller's MTA thread. That could contribute to GC pressure if it's called a lot.
And if it is called a lot, you have a few options:
1. double-down on your expectation that you're always on an MTA anyway and just return `IMoniker` and throw if you're on an STA.
1. return `ValueTask<T>` instead (and in the Task.Run code path wrap the Task with a ValueTask).
1. move your `if` branch logic up to the caller.

Reply:
I choose 3 since the method is not too long. I just inlined it